### ISSUE TITLE
False IgnoredReturnValue positive

### ISSUE THREAD

**Issue by kepocnhh** [detekt/detekt#7460]
## Expected Behavior, Observed Behavior and Steps to Reproduce

If I separate the result of a function into a variable (`val info = storage.getSyncInfo()`), the Detekt is silent.

```kotlin
fun getSyncInfo(hashes: Map<UUID, ByteArray>): Map<UUID, SyncInfo> {
    val result = mutableMapOf<UUID, SyncInfo>()
    for ((id, hash) in hashes) {
        val storage = get(id = id) ?: continue
        val info = storage.getSyncInfo()
        result[id] = info
    }
    return result
}
```

If I pass the result directly to a new object (`result[id] = storage.getSyncInfo()`), then the [IgnoredReturnValue](https://detekt.dev/docs/rules/potential-bugs/#ignoredreturnvalue) rule is triggered for the Detekt.

```kotlin
fun getSyncInfo(hashes: Map<UUID, ByteArray>): Map<UUID, SyncInfo> {
    val result = mutableMapOf<UUID, SyncInfo>()
    for ((id, hash) in hashes) {
        val storage = get(id = id) ?: continue
        result[id] = storage.getSyncInfo()
    }
    return result
}
```

## Environment
* Version of detekt used: 1.23.6
* Version of Gradle used: 8.6
* Kotlin: 1.9.20
* Operating System and version: macOS Sonoma 14.5

**Comment by cortinico** [comment]
This feels like a bug but it should be verified first.

**Comment by kepocnhh** [comment]
If you run a command:
```
gradle lib:checkCodeQuality
```
using the code for this [commit](https://github.com/StanleyProjects/Storages/commit/2bfac818f017a486af5a0dfd161961c9f10ced3a), you can see a message from the Detekt in the report:
<img width="668" alt="image" src="https://github.com/user-attachments/assets/042b33fd-3511-407b-9f64-411a6d8d1e11">

**Comment by cortinico** [comment]
The best would be to add a failing test here:
https://github.com/detekt/detekt/blob/main/detekt-rules-errorprone/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/bugs/IgnoredReturnValueSpec.kt

with your snippet and verify that it actually fails

**Comment by cortinico** [comment]
> Hi @kepocnhh I tried below TC but it is passing. Is there anything different w.r.t. to your example?

Thanks for testing this @atulgpt. nit: should we add it to our test suite?

**Comment by atulgpt** [comment]
Hi @cortinico I have added the TC at https://github.com/detekt/detekt/pull/7541

**Comment by BraisGabin** [comment]
Closing. If a reproducer is provided we can open this again.
